### PR TITLE
Instructor edit rubric question: ignore empty rows #7039

### LINKED ISSUE #7039: Instructor edit rubric question: ignore empty rows
v5.100

Steps:
1. Login as instructor.
2. Edit a rubric question. Leave an empty row in the middle. Try to save.

Actual: All edits are discarded and an error message is given
Suggestion: Either inform about the blank row without discarding all edits, or ignore the empty row and save the rest.

### PR DESCRIPTION
**Pull request by VamsiSangam**
Fixes #7039 

**Outline of Solution**

Added `required` attribute to rubric sub question `textarea` tag. This automatically shows a pop up -

capture

as such when instructor tries to submit an empty sub question.

### CODE REVIEW

**Comment by VamsiSangam** [comment]
Ready for review.

**Comment by damithc** [comment]
> Seems to work for existing rows, but not for new rows (ie. added using the "add row" button) when editing or adding a new rubrics question.

This indicates a gap in tests. There should have been a test to cover this situation.

**Comment by VamsiSangam** [comment]
Ready for review.

File: src/main/resources/feedbackQuestionRubricEditFormBody.html [hunk 1]
```diff
@@ -5,7 +5,7 @@
             onmouseover="highlightRubricRow(${row}, ${questionIndex}, true)" onmouseout="highlightRubricRow(${row}, ${questionIndex}, false)">
                 <span class="glyphicon glyphicon-remove"></span>
             </span>
-            <textarea class="form-control" rows="3" id="${Const.ParamsNames.FEEDBACK_QUESTION_RUBRICSUBQUESTION}-${questionIndex}-${row}" name="${Const.ParamsNames.FEEDBACK_QUESTION_RUBRICSUBQUESTION}-${row}">${subQuestion}</textarea>
+            <textarea class="form-control" rows="3" id="${Const.ParamsNames.FEEDBACK_QUESTION_RUBRICSUBQUESTION}-${questionIndex}-${row}" name="${Const.ParamsNames.FEEDBACK_QUESTION_RUBRICSUBQUESTION}-${row}" required>${subQuestion}</textarea>
```
Comment (chowyb) [inline src/main/resources/feedbackQuestionRubricEditFormBody.html]: Change `required` to `required=""`

File: src/main/webapp/dev/js/common/questionRubric.es6 [hunk 2]
```diff
@@ -45,7 +45,7 @@ function addRubricRow(questionNum) {
                         <span class="glyphicon glyphicon-remove"></span>
                     </span>
                     <textarea class="form-control" rows="3" id="rubricSubQn-${questionNum}-${newRowNumber - 1}"
-                            name="rubricSubQn-${newRowNumber - 1}"></textarea>
+                            name="rubricSubQn-${newRowNumber - 1}" required></textarea>
```
Comment (chowyb) [inline src/main/webapp/dev/js/common/questionRubric.es6]: Change `required` to `required=""` here too

**Pull request merged**